instruments/instruments/abstract_instruments/instrument.py

```diff
@@ -75,39 +71,52 @@
     4: '>i'
 })
 
-## CLASSES #####################################################################
+# CLASSES #####################################################################
+
 
 class Instrument(object):
 
-    # Set a default terminator.
-    # This can and should be overriden in subclasses for instruments
-    # that use different terminators.
-    _terminator = "\n"
-    
     def __init__(self, filelike):
         # Check to make sure filelike is a subclass of AbstractCommunicator
         if isinstance(filelike, AbstractCommunicator):
             self._file = filelike
         else:
             raise TypeError('Instrument must be initialized with a filelike '
-                              'object that is a subclass of AbstractCommunicator.')
-    
-    ## COMMAND-HANDLING METHODS ##
-    
+                            'object that is a subclass of '
+                            'AbstractCommunicator.')
+        self._prompt = None
+        self._terminator = "\n"
+
+    # COMMAND-HANDLING METHODS #
+
+    def _ack_expected(self, msg=""):
+        return None
+
     def sendcmd(self, cmd):
         """
-        Sends a command without waiting for a response. 
-        
+        Sends a command without waiting for a response.
+
         :param str cmd: String containing the command to
             be sent.
         """
         self._file.sendcmd(str(cmd))
-        
+        ack_expected = self._ack_expected(cmd)
+        if ack_expected is not None:
+            ack = self.read()
+            if ack != ack_expected:
+                raise IOError("Incorrect ACK message received: got {} "
```

Comment:
I should make new errors called `AckError` and `PromptError` or something. Thoughts what the correct base they should inherit from @cgranade ?